transformation/src/pipeline-config/model/pipelineConfig.ts

```diff
@@ -16,3 +16,84 @@ export interface Metadata {
   description: string;
   creationTimestamp: Date;
 }
+
+export interface PipelineConfigDTO {
+  datasourceId: number;
+  transformation: TransformationConfig;
+  metadata: MetadataDTO;
+}
+
+export interface MetadataDTO {
+  author: string;
+  displayName: string;
+  license: string;
+  description: string;
+}
+
+export class PipelineConfigDTOValidator {
+  private errors: string[] = []
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  validate (pipelineConfig: any): pipelineConfig is PipelineConfigDTO {
+    this.errors = []
+    if (typeof pipelineConfig !== 'object') {
```

Comment:
The whole `validate()` method is not easy to read. 
You could improve the readability of it be extracting some logic of the IF statements.
E.g.
```js
if (!isObject(pipelineConfig)) {...}
if (!isString(pipelineConfig.someField)) {...}
if (!isNumber(pipelineConfig.someField)) {...}


```